scripts/Muon/GUI/Common/load_run_widget/view.py

```diff
@@ -0,0 +1,189 @@
+
+        self.horizontal_layout.setContentsMargins(0, 0, 0, 0)
+        self.horizontal_layout.setMargin(0)
+
+    def set_item_sizes(self):
+        button_height = self.loadCurrentRunButton.height()
+        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Fixed)
+        sizePolicy.setHeightForWidth(True)
+
+        self.loadCurrentRunButton.setMinimumSize(QtCore.QSize(0, button_height))
+        self.loadCurrentRunButton.setSizePolicy(sizePolicy)
+
+        self.incrementRunButton.setMinimumSize(QtCore.QSize(button_height, button_height))
+        self.incrementRunButton.setSizePolicy(sizePolicy)
+
+        self.decrementRunButton.setMinimumSize(QtCore.QSize(button_height, button_height))
+        self.decrementRunButton.setSizePolicy(sizePolicy)
+
+        self.run_edit.setMinimumSize(QtCore.QSize(50, button_height))
+        self.run_edit.setSizePolicy(QtGui.QSizePolicy(QtGui.QSizePolicy.Preferred, QtGui.QSizePolicy.Fixed))
+
+        self.instrumentLabel.setSizePolicy(sizePolicy)
+
+    def getLayout(self):
+        return self.horizontalLayout
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Enabling / disabling the interface
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def disable_loading(self):
+        self.disable_load_buttons()
+        self.loadingStarted.emit()
+
+    def enable_loading(self):
+        self.enable_load_buttons()
+        self.loadingFinished.emit()
+        self.dataChanged.emit()
+
+    def notify_loading_started(self):
+        self.loadingStarted.emit()
+
+    def notify_loading_finished(self):
+        self.loadingFinished.emit()
+        self.dataChanged.emit()
+
+    def disable_load_buttons(self):
+        self.load_current_run_button.setEnabled(False)
+        self.run_edit.setEnabled(False)
+        self.increment_run_button.setEnabled(False)
+        self.decrement_run_button.setEnabled(False)
+
+    def enable_load_buttons(self):
+        self.load_current_run_button.setEnabled(True)
+        self.run_edit.setEnabled(True)
+        self.increment_run_button.setEnabled(True)
+        self.decrement_run_button.setEnabled(True)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Instrument / run-edit
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def get_instrument_label(self):
+        return str(self.instrument_label.text())
+
+    def set_instrument_label(self, text):
+        self.instrument_label.setText(text)
+
+    def set_current_instrument(self, instrument):
+        self.instrument_label.setText(instrument)
+
+    def set_run_edit_regex(self):
+        # "^[0-9]*([0-9]+[,-]{0,1})*[0-9]+$"
+        regex = QtCore.QRegExp(run_utils.run_string_regex)
```

Comment:
should this be a utils function? Then we can test the RegExp

Reply:
I think the argument for it being in the view is that is uses Qt functionality which we are trying to keep confined to the view. Having said that I think that testing the regex is a good idea I'll add some tests to the view

Reply:
ok